From d16c4eee10a69e772edc45d15c8c6bef618941b4 Mon Sep 17 00:00:00 2001
From: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Date: Wed, 30 Sep 2020 17:58:44 +0530
Subject: [PATCH 1686/1852] pinctrl: zynqmp: Return 'ret' instead of Success
 always

commit 6e094f722ae7c8db077e4dde433f3324f2918217 from
https://github.com/Xilinx/linux-xlnx.git

Return 'ret' for zynqmp_pinconf_cfg_get() function instead of
returning success always.

Addresses-coverity: Parse warning (PW.SET_BUT_NOT_USED).
Signed-off-by: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/pinctrl/pinctrl-zynqmp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/pinctrl/pinctrl-zynqmp.c b/drivers/pinctrl/pinctrl-zynqmp.c
index e7e672afcd0d..8b7d44b32461 100644
--- a/drivers/pinctrl/pinctrl-zynqmp.c
+++ b/drivers/pinctrl/pinctrl-zynqmp.c
@@ -418,11 +418,12 @@ static int zynqmp_pinconf_cfg_get(struct pinctrl_dev *pctldev,
 		}
 		break;
 	default:
-		return -ENOTSUPP;
+		ret = -ENOTSUPP;
+		break;
 	}
 
 	*config = pinconf_to_config_packed(param, arg);
-	return 0;
+	return ret;
 }
 
 /**
-- 
2.31.1

